plugins/automation/app/services/discourse_automation/update.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+class DiscourseAutomation::Update
+  include Service::Base
+
+  # @!method self.call(guardian:, params:)
+  #   @param [Guardian] guardian
+  #   @param [Hash] params
+  #   @option params [Integer] :automation_id
+  #   @option params [String] :name
+  #   @option params [String] :script
+  #   @option params [String] :trigger
+  #   @option params [Boolean] :enabled
+  #   @option params [Array] :fields
+  #   @return [Service::Base::Context]
+
+  policy :can_update_automation
+
+  params do
+    attribute :automation_id, :integer
+    attribute :name, :string
+    attribute :script, :string
+    attribute :trigger, :string
+    attribute :enabled, :boolean
+    attribute :fields, :array
+
+    validates :automation_id, presence: true
+  end
+
+  model :automation
+  step :capture_previous_state
+
+  transaction do
+    step :apply_forced_triggerable
+    step :handle_trigger_change
+    step :handle_script_change
+    step :update_automation
+    step :log_action
+  end
+
+  private
+
+  def can_update_automation(guardian:)
+    guardian.is_admin?
+  end
+
+  def fetch_automation(params:)
+    DiscourseAutomation::Automation.includes(:fields, :pending_automations).find_by(
+      id: params.automation_id,
+    )
+  end
+
+  def capture_previous_state(automation:)
+    context[:previous_state] = {
+      name: automation.name,
+      script: automation.script,
+      trigger: automation.trigger,
+      enabled: automation.enabled,
+      fields: automation.serialized_fields,
+    }
```

Comment:
Following my previous comment, then we can do this:
```suggestion
    automation.slice(:name, :script, :trigger, :enabled).merge(fields: automation.serialized_fields)
```